File: node/packages/aws-lambda-sdk/lib/instrumentation/aws-sdk/safe-stringify.js
'use strict';

const util = require('util');

const replacer = (key, value) => {
  if (typeof value === 'bigint') return value.toString();
  return value;
};

module.exports = (value) => {
  try {
    return JSON.stringify(value, replacer);
  } catch (error) {
    serverlessSdk._reportWarning(
      'Detected not serializable value in AWS SDK request:\n' +
        `\tvalue: ${util.inspect(value)}\n` +
        `\terror:${error.message}`,
      'AWS_SDK_NON_SERIALIZABLE_VALUE'
    );
    return null;
  }
};

const serverlessSdk = require('../../../');

File: python/packages/aws-lambda-sdk/serverless_aws_lambda_sdk/lib/instrumentation/aws_sdk/safe_stringify.py
from sls_sdk.lib.imports import internally_imported

with internally_imported():
    import json

import serverless_aws_lambda_sdk


def safe_stringify(value):
    try:
        return json.dumps(value, default=str)
    except TypeError as ex:
        serverless_aws_lambda_sdk.serverlessSdk._report_warning(
            "Detected not serializable value in AWS SDK request:\n"
            + "\tvalue: {}\n".format(value)
            + "\terror: {}".format(ex),
            "AWS_SDK_NON_SERIALIZABLE_VALUE",
        )
        return None
